### ISSUE TITLE
NullPointerException on com.plugin.gcm.OneSignalPush.onDestroy

### ISSUE THREAD

**Issue by nootherday** [OneSignal/OneSignal-Cordova-SDK#131]
Hi, following error occurred on user devices about 20 times. 

![image](https://cloud.githubusercontent.com/assets/16236022/21087164/9839f024-c066-11e6-9e91-269cb58978b4.png)

My app is based on ionic, OneSignal-Cordova-SDK 2.0.6. I couldn't reproduce it on dev environment. 

```
Fatal Exception: java.lang.RuntimeException: Unable to destroy activity {com.luga/com.luga.MainActivity}: java.lang.NullPointerException: Attempt to write to field 'com.onesignal.OneSignal$NotificationOpenedHandler com.onesignal.OneSignal$Builder.mNotificationOpenedHandler' on a null object reference
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5062)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5085)
       at android.app.ActivityThread.access$1700(ActivityThread.java:221)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1853)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7225)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.NullPointerException: Attempt to write to field 'com.onesignal.OneSignal$NotificationOpenedHandler com.onesignal.OneSignal$Builder.mNotificationOpenedHandler' on a null object reference
       at com.onesignal.OneSignal.removeNotificationOpenedHandler(OneSignal.java:1305)
       at com.plugin.gcm.OneSignalPush.onDestroy(OneSignalPush.java:324)
       at org.apache.cordova.PluginManager.onDestroy(PluginManager.java:298)
       at org.apache.cordova.CordovaWebViewImpl.handleDestroy(CordovaWebViewImpl.java:480)
       at org.apache.cordova.CordovaActivity.onDestroy(CordovaActivity.java:305)
       at android.app.Activity.performDestroy(Activity.java:7102)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1170)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5040)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5085)
       at android.app.ActivityThread.access$1700(ActivityThread.java:221)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1853)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7225)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

How can I resolve this? Thank you in advance.

**Comment by dlaynes** [comment]
I'm having the exact issue. +1

**Comment by dlaynes** [comment]
Related to this issue
https://github.com/OneSignal/OneSignal-Android-SDK/issues/71

**Comment by jkasten2** [comment]
@nootherday It looks like this error is happening when the user closes your app before OneSignal is initialized. We will add a check to our SDK to prevent this in a future update.

To minimize the chances of this happening in the meantime make sure OneSignal `endInit` is called on the first page that is opened with your app.

**Comment by dlaynes** [comment]
@jkasten2 , you are right, this error message shows up if the user does not initialize Onesignal, when the library is already installed in cordova.
Thanks for your help.

**Comment by snowbreeze** [comment]
Facing the same #157

**Comment by jkasten2** [comment]
@nootherday @dlaynes @snowbreeze This is now fixed in the 2.1.0 version of this SDK.
